//line 574
SYSCALL_DEFINE1(chroot, const char __user *, filename)
{
	struct path path;
	int error;

    //Получение пути к дирректории
	error = user_path_dir(filename, &path);
    //Если ошибка то выходим
	if (error)
		goto out;

    Проверить права доступа к данному inode
	error = inode_permission(path.dentry->d_inode, MAY_EXEC | MAY_ACCESS);
    //Если ошибка то выходим
	if (error)
		goto dput_and_out;

	error = -EPERM;
    //Определите, является ли текущая задача выполнимой
	if (!capable(CAP_SYS_CHROOT))
		goto dput_and_out;

    //Заменяет fs->{rootmnt,root} на {mnt,dentry}(новые).
	set_fs_root(current->fs, &path);
	error = 0;
dput_and_out:
	path_put(&path);
out:
	return error;
}